lua/lspconfig/configs.lua

```diff
@@ -139,10 +139,11 @@ function configs.__newindex(t, config_name, config_def)
           M._setup_buffer(client.id, bufnr)
         else
           api.nvim_command(string.format(
-              "autocmd BufEnter <buffer=%d> ++once lua require'lspconfig'[%q]._setup_buffer(%d)"
+              "autocmd BufEnter <buffer=%d> ++once lua require'lspconfig'[%q]._setup_buffer(%d,%d)"
               , bufnr
               , config_name
               , client.id
+              , bufnr
               ))
```

Comment:
Hmm, I must admit that I don't really get why want this `else` block at all. What's the difference between just calling `M._setup_buffer(client.id, bufnr)` directly? Do you know @mjlbach?

Reply:
Pretty sure it's because schedule_wrap doesn't guarantee the function executes while the current buffer is in focus, so if it's not it execute an autocommand on the next BufEnter (which wouldn't trigger if it was already the active buffer)

Reply:
Why do we care about buffer focus at this point though? Especially since we have `bufnr` which would explicitly target the correct buffer regardless what focused buffer is.

Reply:
I didn't write it, so I don't know the full motivation. One advantage is it differs attaching the client, and running on_attach, until you actually enter the buffer, so if a user adds buffer local keymaps without specifying the buffer (so using the current buffer), it "just works"

Reply:
That's a valid point. Especially if you have a habit of doing something like this:

```console
$ nvim src/**.py
```

Reply:
True, and somebody probably uses that [behavior](https://www.hyrumslaw.com/).